src/main/java/com/watch/shop/app/Main.java

```diff
@@ -1,4 +1,19 @@
 package com.watch.shop.app;
 
+import com.watch.shop.app.controller.MenuHandler;
+import com.watch.shop.app.controller.WatchController;
+import com.watch.shop.app.model.service.WatchService;
+import com.watch.shop.app.model.service.WatchServiceImpl;
+import com.watch.shop.app.view.WatchView;
+
 public class Main {
+
+    public static void main(String[] args) {
+        WatchService service = new WatchServiceImpl();
+        WatchView view = new WatchView();
+        MenuHandler menuHandler = new MenuHandler(view);
+        WatchController controller = new WatchController(service, view, menuHandler);
```

Comment:
We have nesting dependency:
View
Controller
  View
  MenuHandler
     View
     
..
 Since MenuHandler is helper class, it's 1 level lower than View, InputHandler, Service, Controller
 
 So you could just make constructor for Controller:
  puvblic Controller(WatchView view, InputHandler inputHandler, WatchService service) {
   this.view = view;
   ..
    this.menuHandler = new MenuHandler(view);
   }
   
   In this way will be little bit less mess with injecting View dependency

Reply:
I wanted to do it, but I really like annotations, so I decided not to do it that time 😄
I've corrected it now